packages/Meshfree/test/tstMeshfreeComparison.cpp

```diff
@@ -0,0 +1,271 @@
+            out_host( i ) = in[i];
+        Kokkos::deep_copy( out, out_host );
+        return out;
+    }
+};
+
+template <typename Operator>
+void testOperator( int source_points_per_dim, int target_points_per_dim )
+{
+    using DeviceType = typename Operator::device_type;
+
+    MPI_Comm comm = MPI_COMM_WORLD;
+    int comm_rank;
+    MPI_Comm_rank( comm, &comm_rank );
+
+    int comm_size;
+    MPI_Comm_size( comm, &comm_size );
+
+    int l = std::ceil( std::cbrt( comm_size ) );
+
+    double offset_x = ( comm_rank % l ) * 1. / l;
+    double offset_y = ( ( comm_rank / l ) % l ) * 1. / l;
+    double offset_z = ( ( comm_rank / ( l * l ) ) % l ) * 1. / l;
+
+    std::array<int, DIM> n_source_points_grid = {
+        source_points_per_dim, source_points_per_dim, source_points_per_dim};
+    std::array<double, DIM> lower_corner = {offset_x, offset_y, offset_z};
+    std::array<double, DIM> upper_corner = {
+        offset_x + 1. / l, offset_y + 1. / l, offset_z + 1. / l};
+    auto source_points_arr = Helper<DeviceType>::makeGridPoints(
+        n_source_points_grid, lower_corner, upper_corner );
+
+    std::array<int, DIM> n_target_points_grid = {
+        target_points_per_dim, target_points_per_dim, target_points_per_dim};
+    auto target_points_arr = Helper<DeviceType>::makeGridPoints(
+        n_target_points_grid, lower_corner, upper_corner );
+
+    unsigned int const n_source_points = source_points_arr.size();
+    unsigned int const n_target_points = target_points_arr.size();
+    std::vector<double> source_values_arr( n_source_points );
+    std::vector<double> target_values_arr( n_target_points );
+    std::vector<double> target_values_ref( n_target_points );
+
+    // Arbitrary function of the specified order
+    std::function<double( std::array<double, DIM> )> f =
+        []( std::array<double, DIM> p ) -> double {
+        return 2 + 3 * p[0] - 5 * p[1] + 2 * p[2] + 3 * p[0] * p[0] +
+               4 * p[0] * p[1] - 2 * p[0] * p[2] + p[1] * p[1] -
+               3 * p[1] * p[2] + 4 * p[2] * p[2] -
+               std::sin( p[0] ) * std::sin( p[1] ) * std::sin( p[2] );
+    };
+
+    for ( unsigned int i = 0; i < n_source_points; ++i )
+        source_values_arr[i] = f( source_points_arr[i] );
+    for ( unsigned int i = 0; i < n_target_points; ++i )
+        target_values_ref[i] = f( target_points_arr[i] );
+
+    auto source_points = Helper<DeviceType>::makePoints( source_points_arr );
+    auto source_values = Helper<DeviceType>::makeValues( source_values_arr );
+    auto target_points = Helper<DeviceType>::makePoints( target_points_arr );
+    auto target_values = Helper<DeviceType>::makeValues( target_values_arr );
+
+    std::cout << "(" << source_points_per_dim << "," << target_points_per_dim
+              << ") ";
+
+    MPI_Barrier( MPI_COMM_WORLD );
+    Kokkos::fence();
+    auto start_setup = std::chrono::high_resolution_clock::now();
+    Operator op( comm, source_points, target_points );
```

Comment:
I would probably iterate several times and compute average or median just to remove outliers.